privacyidea/lib/smsprovider/SmppSMSProvider.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+#
+#    E-mail: yurlov.alexandr@gmail.com
+#
+#    2017-12-22 Alexander Yurlov <yurlov.alexandr@gmail.com>
+#               Add new SMPP SMS provider
+#
+#
+#    Copyright (C) LinOTP: 2010 - 2014 LSE Leading Security Experts GmbH
+#
+#    This program is free software: you can redistribute it and/or
+#    modify it under the terms of the GNU Affero General Public
+#    License, version 3, as published by the Free Software Foundation.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the
+#               GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+__doc__="""This is the SMSClass to send SMS via SMPP protocol to SMS center
+It requires smpplib installation, this lib works with ascii only, but message support unicode 
+
+"""
+
+from privacyidea.lib.smsprovider.SMSProvider import (ISMSProvider, SMSError)
+import requests
+from urlparse import urlparse
+
+import logging
+log = logging.getLogger(__name__)
+
+import smpplib
```

Comment:
The python smpplib is currently not available on Ubuntu. 
In the long run we problably need to package this.
Till then I recommend wrapping the import into an exception, so that we will not fail on ubuntu.

Reply:
@cornelinux Are you sure python smpplib not available on Ubuntu?
I just installed it from command line on ubuntu 16.04 xenial.
```
# pip install smpplib
Collecting smpplib
  Downloading smpplib-0.1.tar.gz
Building wheels for collected packages: smpplib
  Running setup.py bdist_wheel for smpplib ... done
  Stored in directory: /root/.cache/pip/wheels/18/15/0f/55054fd26dcb99c0721b0d5f1238b45a98461e799d295317f6
Successfully built smpplib
Installing collected packages: smpplib
Successfully installed smpplib-0.1
#
```

Reply:
Yes. I am sure.
When talking of ubuntu I mean deb packages in the ubuntu repo. These are not available.
Installing privacyIDEA from the ubuntu/launchpad repositories would then fail, because smpplib is not available as ubuntu package.

Reply:
Indeed, I use it on CentOS 7. I installed privacyIDEA manually with this guide https://www.privacyidea.org/two-factor-authentication-with-otp-on-centos-7/
smpplib was installed via pip